### PR TITLE
Updated the docs to split it in smaller chapters and use Flex

### PR DESCRIPTION
**Pull request by javiereguiluz**
This replaces #268 and finishes the upgrade of docs.

### CODE REVIEW

File: Resources/doc/index.rst [hunk 1]
```diff
@@ -1,356 +1,43 @@
...
-  version management.
-- **Sortable** - makes any document or entity sortable
-- **Translator** - explicit way to handle translations
-- **Softdeleteable** - allows to implicitly remove records
-- **Uploadable** - provides file upload handling in entity fields
-- **Reference Integrity** - provides reference integrity for MongoDB, supports
+* **Tree** - this extension automates the tree handling process and adds some
+  tree specific functions on repository. (``closure``, ``nestedset`` or ``materialized path``).
+* **Translatable** - gives you a very handy solution for translating records
+  into diferent languages. Easy to setup, easier to use.
```
Comment (stof) [inline Resources/doc/index.rst]: typo: `different` (and it was not there in the removed code)

File: Resources/doc/installation.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
+Installation
+============
+
+Installation Using Symfony Flex
+-------------------------------
+
+`Symfony Flex`_ is the new way to manage dependencies on Symfony 3.3 and higher
```
Comment (stof) [inline Resources/doc/installation.rst]: I would say 3.4 here, as using Flex with SF 3.3 is not entirely recommended (it does not work as good as in 3.4+). I don't want to push people to Flex on 3.3 in this bundle (they can still do it, but this doc should not be responsible for it 😄)

File: Resources/doc/softdeleteable-filter.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
+SoftDeletable Filter
```
Comment (stof) [inline Resources/doc/softdeleteable-filter.rst]: should have the `e` in it, to be consistent with the extension name IMO

**Comment by Oliboy50** [comment]
@stof @javiereguiluz what's the status of this PR?

it would be nice to merge it asap, because i'd like to add documentation about using `timestampable` with traits to fix this issue: https://github.com/stof/StofDoctrineExtensionsBundle/issues/367

**Pull request merged**